### PR TITLE
Replace deprecated qt5_use_modules function

### PR DESCRIPTION
**Pull request by revolter**
I'm more curious for the result of a Travis build, so if it fails and it doesn't make sense, feel free to close this.

### CODE REVIEW

**Comment by MKleusberg** [comment]
Hmm I would guess it's ```Qt5Network``` or something like that?

**Comment by MKleusberg** [comment]
Oh, it's probably ```Qt5Xml``` or something similar.

**Comment by revolter** [comment]
@MKleusberg, That's already added here: https://github.com/revolter/sqlitebrowser/blob/5e1c40fe4dbb058524721077db715afc5554bb08/CMakeLists.txt#L85

**Comment by revolter** [comment]
The fail is here: https://travis-ci.org/sqlitebrowser/sqlitebrowser/jobs/389362594#L1197

File: CMakeLists.txt [hunk 1]
```diff
@@ -366,6 +366,9 @@ if(NOT QSCINTILLA_FOUND)
     link_directories("${CMAKE_CURRENT_BINARY_DIR}/${QSCINTILLA_DIR}")
 endif()
 
+message("find me")
```
Comment (justinclift) [inline CMakeLists.txt]: Found it!  Do I win a prize? 😁

Comment (iKlsR) [reply CMakeLists.txt]: Haha, just some trial and error with Travis

**Comment by iKlsR** [comment]
This is going to merit some discussion because the current CMakeLists is pretty archaic and filled with cruft. We should start to use some more modern features that CMake offers instead of sticking to the "safe" 2.8.

*Not sure what the end goal of this PR is tbh ;)*... but for those who need to keep up with modern Qt versions, using deprecated features isn't always an option so we bump the minimum required CMake version, prefix some includes that are apart of a larger module and slim the libs to target a little.

**Comment by justinclift** [comment]
With the CMake version, we're still supporting people compiling on Kubuntu 14.04 (#1298), so we'd probably want to check the version of CMake available to that.

That being said, I'm not sure if people using Kubuntu are compiling with CMake or Qmake.  Either should (in theory) work well enough to produce an executable.

Note - From memory, we haven't ever gotten around to hooking up support files (eg translation files) with Qmake.

**Comment by justinclift** [comment]
@GortiZ6 When you compile DB4S on your Kubuntu 14.04 system, are you doing it with `cmake` or `qmake`?  Just checking as we're wanting to adjust the minimum CMake version we use, so are doing some info collection. :smile:

**Comment by GortiZ6** [comment]
Hi @justinclift, I was using the `CMake` due to one of my first commit being "rejected" due to an error not snown in `QMake`, but in `CMake` only... I don't remember the exact reason (maybe I simply forgot to add files in the CMakeList.txt). Anyway if `qmake` builds, I don't need to have `cmake` compatibility. Furthermore we are also migrating away from KUbuntu 14.04 (which will be out of support next year) and moving to KUbuntu 16.04/18.04 so no problem for me.

**Comment by justinclift** [comment]
Thanks @GortiZ6. :smile:

**Comment by iKlsR** [comment]
Ongoing discussion and testing in our Slack, feel free to drop by. 

[![Join the chat at https://join.slack.com/t/db4s/shared_invite/enQtMzc3MzY5OTU4NDgzLWRlYjk0ZmE5ZDEzYWVmNDQxYTYxNmJjNWVkMjI3ZmVjZTY2NDBjODY3YzNhNTNmZDVlNWI2ZGFjNTk5MjJkYmU][slack-img]][slack]

[slack-img]: https://img.shields.io/badge/chat-on%20slack-orange.svg
  [slack]: https://join.slack.com/t/db4s/shared_invite/enQtMzc3MzY5OTU4NDgzLWRlYjk0ZmE5ZDEzYWVmNDQxYTYxNmJjNWVkMjI3ZmVjZTY2NDBjODY3YzNhNTNmZDVlNWI2ZGFjNTk5MjJkYmU

**Comment by revolter** [comment]
Trying to build it locally fails as I said:

```
> cmake --version
cmake version 3.11.2

> cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -DENABLE_TESTING=ON ..
CMake Error at libs/qscintilla/Qt4Qt5/CMakeLists.txt:170 (qt5_use_modules):
  Unknown CMake command "qt5_use_modules".


-- Configuring incomplete, errors occurred!
See also "/Users/revolt/Development/GitHub/revolter/sqlitebrowser/build/CMakeFiles/CMakeOutput.log".
```

**Comment by revolter** [comment]
So it works with `CMake 3.9.2`, but it doesn't work for me locally with `CMake 3.11.1`.

**Comment by FriedrichFroebel** [comment]
From what I have seen, this is the problem reported in https://github.com/sqlitebrowser/sqlitebrowser/issues/1361.

**Comment by justinclift** [comment]
Yeah, makes sense. :smile:

**Comment by mgrojo** [comment]
This branch is working for me.

Ubuntu 16.04
cmake version 3.5.1
Qt Version 5.5.1

**Comment by justinclift** [comment]
Hmmm, #1421 itself is resolved now... does that have an effect on this?

**Comment by GortiZ6** [comment]
Works also for me:
Ubuntu 18.04
cmake version 3.10.2
Qt Version 5.9.5

**Comment by MKleusberg** [comment]
It works also for me - on cmake and on qmake. As everybody else is reporting a success as well, I'll go ahead and merge this. Thanks again, @revolter :smile:

**Comment by revolter** [comment]
Wow, didn't think this would get merged. Many thanks to @iKlsR as he did the hardest part!

**Pull request merged**